spec/services/catalog/order_item_sanitized_parameters_spec.rb

```diff
@@ -75,29 +36,95 @@
     class_double(TopologicalInventory).as_stubbed_const(:transfer_nested_constants => true)
   end
 
-  before do
-    allow(topological_inventory).to receive(:call).and_yield(api_instance)
-    allow(api_instance).to receive(:show_service_plan).and_return(plan)
-    allow(plan).to receive(:create_json_schema).and_return(json_schema)
-  end
+  describe "#process" do
+    before do
+      allow(topological_inventory).to receive(:call).and_yield(api_instance)
+      allow(api_instance).to receive(:show_service_plan).and_return(plan)
+      allow(plan).to receive(:create_json_schema).and_return(json_schema)
+    end
 
-  context "#process" do
-    it "success scenario" do
-      expect(api_instance).to receive(:show_service_plan)
-        .with(order_item.service_plan_ref)
-        .and_return(plan)
+    context "ocp parameters" do
+      let(:json_schema) do
+        {
+          :properties => {
+            'user'     => {:title => 'User Name', :type => 'string'},
+            'password' => {:title => 'Secret', :type => 'string'},
+            'newpwd'   => {:title => 'Users Password', :type => 'string'},
+            'salary'   => {:title => 'Salary', :format => 'password', :type => 'number'},
+            'db_name'  => {:title => 'Database Name', :type => 'string'}
+          }
+        }
+      end
 
-      result = oisp.process
+      it "success scenario" do
+        expect(api_instance).to receive(:show_service_plan)
+          .with(order_item.service_plan_ref)
+          .and_return(plan)
 
-      expect(result.values.select { |v| v == described_class::MASKED_VALUE }.count).to eq(3)
+        result = oisp.process
+
+        expect(result.values.select { |v| v == described_class::MASKED_VALUE }.count).to eq(3)
+      end
+
+      it "failure scenario" do
+        expect(api_instance).to receive(:show_service_plan)
+          .with(order_item.service_plan_ref)
+          .and_raise(TopologicalInventoryApiClient::ApiError.new("Kaboom"))
+
+        expect { oisp.process }.to raise_error(StandardError)
+      end
     end
 
-    it "failure scenario" do
-      expect(api_instance).to receive(:show_service_plan)
-        .with(order_item.service_plan_ref)
-        .and_raise(TopologicalInventoryApiClient::ApiError.new("Kaboom"))
+    context "ansible parameters" do
+      let(:json_schema) do
+        {
+          :schema => {
+            :fields => [
+              {
+                :name         => "Totally not a password",
```

Comment:
Technically since this contains "password", it will still potentially match the regex. Get rid of the word "password" here and in the label.